crates/pathfinder/src/storage/state.rs

```diff
@@ -567,6 +567,38 @@ impl StarknetTransactionsTable {
         Ok(data)
     }
 
+    pub fn get_transactions_for_latest_block(
+        sqlite_tx: &Transaction<'_>,
+    ) -> anyhow::Result<Vec<transaction::Transaction>> {
+        let block_hash = match StarknetBlocksTable::get(sqlite_tx, StarknetBlocksBlockId::Latest)? {
+            Some(block) => block.hash,
+            None => return Ok(Vec::new()),
+        };
+
+        let mut stmt = sqlite_tx
+            .prepare("SELECT tx FROM starknet_transactions WHERE block_hash = ? ORDER BY idx ASC")
+            .context("Preparing statement")?;
```

Comment:
This could be:

```
     SELECT txs.tx
       FROM starknet_transactions txs
       JOIN starknet_blocks bs ON (bs.block_hash = txs.block_hash)
      WHERE bs.number = (SELECT max(number) from starknet_blocks)
   ORDER BY txs.idx ASC
```

Or something similar. It'd avoid the unnecessary extra query.